lib/webmock/http_lib_adapters/manticore_adapter.rb

```diff
@@ -0,0 +1,104 @@
+begin
+  require 'manticore'
+rescue LoadError
+  # manticore not found
+end
+
+if defined?(Manticore)
+  module WebMock
+    module HttpLibAdapters
+      class ManticoreAdapter < HttpLibAdapter
+        adapter_for :manticore
+
+        OriginalManticoreClient = Manticore::Client
+
+        def self.enable!
+          Manticore.send(:remove_const, :Client)
+          Manticore.send(:const_set, :Client, WebMockManticoreClient)
+        end
+
+        def self.disable!
+          Manticore.send(:remove_const, :Client)
+          Manticore.send(:const_set, :Client, OriginalManticoreClient)
+        end
+
+        class WebMockManticoreClient < Manticore::Client
+          def http(method, uri, options = {})
+            @method = method
+            @uri = uri
+            @options = options
+            super(method, WebMock::Util::URI.normalize_uri(uri).to_s, format_options(options))
+          end
+
+          private
+
+          def format_options(options)
+            return options unless headers = options[:headers]
+
+            options.merge(:headers => join_array_values(headers))
+          end
+
+          def join_array_values(headers)
+            headers.reduce({}) do |h, (k,v)|
+              v = v.join(', ') if v.is_a?(Array)
+              h.merge(k => v)
+            end
+          end
+
+          def response_object_for(client, request, context, &block)
```

Comment:
This is a private method that doesn't seem to be called anywhere; is this correct?